app/kubemci/pkg/validations/validations.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validations
+
+import (
+	"fmt"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/kubeutils"
+	"github.com/golang/glog"
+	multierror "github.com/hashicorp/go-multierror"
+	"k8s.io/api/extensions/v1beta1"
+)
+
+func Validate(clients map[string]kubeclient.Interface, ing *v1beta1.Ingress) error {
+	return servicesNodePortsSame(clients, ing)
+}
+
+// ServicesNodePortsSame checks that for each backend/service, the services are
+// all listening on the same NodePort.
+func servicesNodePortsSame(clients map[string]kubeclient.Interface, ing *v1beta1.Ingress) error {
+	var multiErr error
+	for _, rule := range ing.Spec.Rules {
+		if rule.HTTP == nil {
+			glog.Warningf("ignoring non http Ingress rule: %v", rule)
+			continue
+		}
+		for _, path := range rule.HTTP.Paths {
+			glog.Infof("Validating path: %s", path.Path)
+			if err := nodePortSameInAllClusters(path.Backend, ing.Namespace, clients); err != nil {
+				multierror.Append(multiErr, fmt.Errorf("nodePort validation error for service '%s/%s': %s", ing.Namespace, path.Backend.ServiceName, err))
+			}
+		}
+	}
+	glog.Infof("Checking default backend's nodeports.")
+
+	if ing.Spec.Backend != nil {
+		if err := nodePortSameInAllClusters(*ing.Spec.Backend, ing.Namespace, clients); err != nil {
+			glog.Errorf("nodePort validation error for default backend service '%s/%s': %s", *ing.Spec.Backend, ing.Namespace, err)
```

Comment:
fmt.Printf.
Also keep this consistent with error for non backend service. I see we are not printing the error there